src/Network/NWProtocolDefinition.cs

```diff
@@ -44,28 +44,75 @@ public bool Equals (object other)
 
+		public static NWProtocolDefinition CreateIPDefinition () => new NWProtocolDefinition (nw_protocol_copy_ip_definition (), owns: true);
+
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_definition nw_protocol_copy_tcp_definition ();
 
+		[Obsolete ("Use 'CreateTcpDefinition' method instead.")]
 		public static NWProtocolDefinition TcpDefinition => new NWProtocolDefinition (nw_protocol_copy_tcp_definition (), owns: true);
 
+		public static NWProtocolDefinition CreateTcpDefinition () => new NWProtocolDefinition (nw_protocol_copy_tcp_definition (), owns: true);
+
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_definition nw_protocol_copy_udp_definition ();
 
+		[Obsolete ("Use 'CreateUdpDefinition' method instead.")]
 		public static NWProtocolDefinition UdpDefinition => new NWProtocolDefinition (nw_protocol_copy_udp_definition (), owns: true);
 
+		public static NWProtocolDefinition CreateUdpDefinition () => new NWProtocolDefinition (nw_protocol_copy_udp_definition (), owns: true);
+
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_definition nw_protocol_copy_tls_definition ();
 
+		[Obsolete ("Use 'CreateTlsDefinition' method instead.")]
 		public static NWProtocolDefinition TlsDefinition => new NWProtocolDefinition (nw_protocol_copy_tls_definition (), owns: true);
 
+		public static NWProtocolDefinition CreateTlsDefinition () => new NWProtocolDefinition (nw_protocol_copy_tls_definition (), owns: true);
+
 		[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
 		[DllImport (Constants.NetworkLibrary)]
 		static extern OS_nw_protocol_definition nw_protocol_copy_ws_definition ();
 
+		[Obsolete ("Use 'CreateWebSocketDefinition' method instead.")]
 		[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
 		public static NWProtocolDefinition WebSocketDefinition => new NWProtocolDefinition (nw_protocol_copy_ws_definition (), owns: true);
+
+		[TV (13,0), Mac (10,15), iOS (13,0), Watch (6,0)]
+		public static NWProtocolDefinition CreateWebSocketDefinition () => new NWProtocolDefinition (nw_protocol_copy_ws_definition (), owns: true);
+
+		[Watch (6,0), TV (13,0), Mac (10,15), iOS (13,0)]
+		[DllImport (Constants.NetworkLibrary)]
+		static extern unsafe OS_nw_protocol_definition nw_framer_create_definition (string identifier, NWFramerCreateFlags flags, ref BlockLiteral start_handler);
+		delegate NWFramerStartResult nw_framer_create_definition_t (IntPtr block, IntPtr framer);
+		static nw_framer_create_definition_t static_CreateFramerHandler = TrampolineCreateFramerHandler;
+
+		[MonoPInvokeCallback (typeof (nw_framer_create_definition_t))]
+		static NWFramerStartResult TrampolineCreateFramerHandler (IntPtr block, IntPtr framer)
+		{
+			// get and call, this is internal and we are trying to do all the magic in the call
+			var del = BlockLiteral.GetTarget<Func<NWFramer,NWFramerStartResult>> (block);
+			if (del != null) {
+				var nwFramer = new NWFramer (framer, owns: true);
```

Comment:
That has to be managed by the user. The callback will provide the framer to the user, the framer is later used to read and write message that are sent to the NWConnection, so you want to create a framer per protocol definition, keep it in memory and use it during the life of the application or the usefulness of the object. Then be disposed.

If I do a dispose here, the entire API is useless, since you won't the able to write the required data to the 'wire'.

Reply:
ok, sounds like it's not called often